docs/workload-development-kit/fabric-data-plane.md

```diff
@@ -12,41 +12,98 @@ ms.date: 05/21/2024
 
 ## How to read and write data in Microsoft Fabric 
 
 Microsoft Fabric is a platform that enables the storage and management of customer data. In order to read and write data in Fabric, you need to use the [Fabric REST APIs](/rest/api/fabric/articles/) and the appropriate authentication methods.
 
+### API Permissions
+
+Some methods of accessing customer data require the use of other services outside of Fabric such as Azure Storage or Azure SQL Database. For example, in the Microsoft Fabric Developer kit sample, the API permission Azure Storage user_impersonation is used in conjunction with the PowerBI Service Lakehouse.Read.All permission in order to access data from Lakehouses.
+
+You may choose to use Azure SQL Database in order to access table data from Warehouse items. In this case you would configure your app with Azure SQL Database user_impersonation in order to query the database on behalf of the user and PowerBI Service Warehouse.Read.All in order to query the Fabric REST API Get Warehouse endpoint.
+
+Make sure you configure your Microsoft Entra ID app according to your development needs.
+
 ### Authentication 
 
-Before you can begin using the Fabric REST APIs, you need to authenticate using a token. This token can be obtained through a token exchange process.
+Before you can begin using the Fabric REST APIs or other services, you need to authenticate using a token. This token can be obtained through a token exchange process.
+
+The Fabric Workload Development Kit SDK provides a method for acquiring an access token in the workload frontend. An example of usage can be found in the frontend [Sample Workload Controller](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/src/controller/SampleWorkloadController.ts).
+
+ This client token must be passed to the workload backend and exchanged using the on-behalf-of flow for a token with the necessary scopes to access your desired resource, such as OneLake. For example, in order to access and read from a Lakehouse, a user must authorize the application to make API calls on their behalf using the Azure Storage user_impersonation permission. Then, in the backend, the access token must be obtained with the delegated scope `https://storage.azure.com/user_impersonation` in order to use Azure Storage.
+
+If you decide to use SQL to access your customer data, the access token must be obtained with the scope `https://database.windows.net//user_impersonation` in order to use Azure SQL Database and the Microsoft.Data.SqlClient namespace. The access token can be obtained without the extra forward slash in the scope from Azure SQL Database, *however* the access token won't be validated by the SQLConnection class, and so we recommend using the scope as written here.
```

Comment:
*Microsoft.Data.SqlClient*

Instead of this:

'The access token can be obtained without the extra forward slash in the scope from Azure SQL Database, *however* the access token won't be validated by the SQLConnection class, and so we recommend using the scope as written here.'

Tell the customer to use the scope as written (i.e. the right way that will not cause problems).